### PR TITLE
APPS-4250: Fixed executable path.

### PR DESCRIPTION
**Pull request by stereomon**
- Developer(s): @vol4onok

- Ticket: https://spryker.atlassian.net/browse/APPS-4250

- Release Group: https://release.spryker.com/release-groups/view/4151

- PR Overview: https://release.spryker.com/release/package?repo=spryker-sdk%2Fasync-api&branch=bugfix%2Fapps-4250-fix-executable-path-in-asyncapi-sdk&pr=6

- merge: squash

Module AsyncApi

##### Change log

Improvements

- Removed `AsyncApiConfig::getRootPath()`.
- Introduced `AsyncApiConfig::getSprykRunExecutablePath()`.
- Introduced `AsyncApiInfo::addedPropertyWithTypeTo()`.
- Introduced `AsyncApiInfo::addedTransferDefinitionTo()`.
- Introduced `AsyncApiInfo::addedMessageHandlerPluginForMessageTo()`.
- Adjusted `AsyncApiCodeBuilder` so it uses new path for spryk command from config instead of project path.
- Adjusted namespace for transfers.

### CODE REVIEW

File: src/Transfer/AsyncApiMessageTransfer.php [hunk 1]
```diff
@@ -695,9 +695,9 @@ public function setChannelOrFail(AsyncApiChannelTransfer $channel)
     /**
      * @module AsyncApi
      *
-     * @throws \Spryker\Shared\Kernel\Transfer\Exception\NullValueException
+     * @return \Transfer\AsyncApiChannelTransfer
+     *@throws \Spryker\Shared\Kernel\Transfer\Exception\NullValueException
```
Comment (stereomon) [inline src/Transfer/AsyncApiMessageTransfer.php:699]: Please check

File: src/Transfer/AsyncApiRequestTransfer.php [hunk 2]
```diff
@@ -395,11 +395,11 @@ public function getAsyncApi()
     /**
      * @module AsyncApi
      *
-     * @param \Generated\Shared\Transfer\AsyncApiTransfer $asyncApi
-     *
-     * @throws \Spryker\Shared\Kernel\Transfer\Exception\NullValueException
+     * @param \Transfer\AsyncApiTransfer $asyncApi
      *
      * @return $this
+     *@throws \Spryker\Shared\Kernel\Transfer\Exception\NullValueException
```
Comment (stereomon) [inline src/Transfer/AsyncApiRequestTransfer.php:401]: Here as well. Was the CS sniffer not running?

Comment (stereomon) [reply src/Transfer/AsyncApiRequestTransfer.php:401]: Can you also replace this exception, please?

File: src/SprykerSdk/Shared/AsyncApi/Transfer/asyncapi.transfer.xml [hunk 3]
```diff
@@ -1,67 +0,0 @@
-<?xml version="1.0"?>
```
Comment (stereomon) [inline src/SprykerSdk/Shared/AsyncApi/Transfer/asyncapi.transfer.xml:1]: This file must not be removed, please only rename it so the TransferFinder can't find this file anymore.

`src/SprykerSdk/Transfer/asyncapi.xml`

File: src/Transfer/AsyncApiMessageTransfer.php (hunk 1 above)
Comment (stereomon) [reply src/Transfer/AsyncApiMessageTransfer.php:699]: Don't you see my comments?

File: src/Transfer/AsyncApiRequestTransfer.php (hunk 2 above)
Comment (stereomon) [reply src/Transfer/AsyncApiRequestTransfer.php:401]: Please check all the places.

File: composer.json [hunk 4]
```diff
@@ -6,10 +6,10 @@
     "require": {
         "php": ">=7.4",
         "spryker-sdk/spryk": "^0.4.0",
-        "symfony/console": "^5.3 || ^6.0",
-        "symfony/finder": "^5.3 || ^6.0",
-        "symfony/process": "^5.4",
-        "symfony/property-access": "^5.4"
+        "symfony/console": "^4.0.0 || ^5.3 || ^6.0",
```
Comment (stereomon) [inline composer.json:9]: Please update ci.yml so we have a prefer-lowest check when using PHP7.4

**Review by miroshnikov-spryker — APPROVED** [review]

**Pull request merged**